test/unit/file.test.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * sonar-scanner-npm
+ * Copyright (C) 2022-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import fs from 'fs';
+import * as fsExtra from 'fs-extra';
+import path from 'path';
+import AdmZip from 'adm-zip';
+import zlib from 'zlib';
+import tarStream from 'tar-stream';
+import { extractArchive, getCachedFileLocation } from '../../src/file';
+import { log, LogLevel } from '../../src/logging';
+import { SONAR_CACHE_DIR } from '../../src/constants';
+import { Readable } from 'stream';
+
+// Mock the filesystem
+jest.mock('fs', () => ({
+  createReadStream: jest.fn().mockImplementation(() => {
+    const mockStream = new Readable({
+      read() {
+        process.nextTick(() => this.emit('end')); // emit 'end' on next tick
+      },
+    });
+    mockStream.pipe = jest.fn().mockReturnThis();
+    return mockStream;
+  }),
+  createWriteStream: jest.fn(),
+  existsSync: jest.fn(),
+}));
+
+jest.mock('fs-extra', () => ({}));
+
+jest.mock('adm-zip', () => {
+  const MockAdmZip = jest.fn();
+  MockAdmZip.prototype.extractAllTo = jest.fn();
+  return MockAdmZip;
+});
+
+jest.mock('tar-stream', () => ({
+  extract: jest.fn().mockReturnValue({
+    on: jest.fn(),
+  }),
+}));
+
+jest.mock('zlib', () => ({
+  createGunzip: jest.fn(),
+}));
+
+describe('extractArchive', () => {
+  it('should extract zip files to the specified directory', async () => {
+    const archivePath = 'path/to/archive.zip';
+    const extractPath = 'path/to/extract';
+
+    await extractArchive(archivePath, extractPath);
+
+    const mockAdmZipInstance = (AdmZip as jest.MockedClass<typeof AdmZip>).mock.instances[0];
+    expect(mockAdmZipInstance.extractAllTo).toHaveBeenCalledWith(extractPath, true);
+  });
+
+  it('should extract tar.gz files to the specified directory', async () => {
+    //TODO: think if a good way to test this
+  });
```

Comment:
I would drop this, have a subtask for it or implement it (see suggestion for tar-stream mock above)

Have you considered not mocking at all and actually extracting an archive for testing? I think both options would be acceptable here (avoid lot of mocking and tests end-to-end but breaks test isolation)

Reply:
I will create a subtask to test this. 

I was looking into how these libs test and that's what they do, run the extraction, and then a cleanup. The problem I see is that we are then testing lib code that we don't own and isn't our responsibility.

On the other hand, it would catch an issue that would come from a version bump on these libs 🤷🏻 .

I'll add these comments to the ticket 👍🏻